src/librbd/BlockGuard.h

```diff
@@ -81,8 +81,16 @@ class BlockGuard {
       // request against an already detained block
       detained_block_extent = &(*it);
       if (block_operation != nullptr) {
+	m_detained_block_extents.erase(it);
         detained_block_extent->block_operations.emplace_back(
           std::move(*block_operation));
+
+	if (block_extent.block_start < detained_block_extent->block_extent.block_start)
+	  detained_block_extent->block_extent.block_start = block_extent.block_start;
+	if (block_extent.block_end > detained_block_extent->block_extent.block_end)
+	  detained_block_extent->block_extent.block_end = block_extent.block_end;
+
+	m_detained_block_extents.insert(*detained_block_extent);
```

Comment:
indentation -- this file doesn't use tabs

Reply:
i met this indentation problem once more. But how to avoid this? and how to see indentation by tabs or space? Thanks!

Reply:
Just check the surrounding lines in your editor -- if there are no tabs, don't introduce them.  Unfortunately we have a mix of tab+spaces and spaces-only files...